consul-topology/go.mod

```
@@ -0,0 +1,44 @@
+module github.com/hashicorp/consul/consul-topology
```

```suggestion
module github.com/hashicorp/consul/test/topology
```

This would also require moving the whole tree underneath the test directory.

---

That could work. It's a bit incongruous that `test/topology` means "the test framework for deploying topologies" whereas `test/integration` means "integration tests" but I could live with that.

It will be weird when we need to use `Config`, which would be 

```
import github.com/hashicorp/consul/test/topology/topology
```

Maybe we pull the contents of `test/topology/topology` into just `test/topology`.

---

If we wanted better distinction between frameworks and actual tests we could have the module be in test/frameworks/topology. 

I don't have strong opinions against putting the topology related tests into test/topology either. The main downside I see there is that we will need more CI jobs to run both versions of tests whereas I think if all the integ tests lived in test/integration/consul-container module then we wouldn't have to do much if anything to make them run. Maybe thats not desirable and we want to distinguish them in CI?

Also this will all require a bunch of import modifications. If you didn't know it existed I have a tool to do import rewriting:

```
# Install the tool
go install github.com/mkeeler/goimport-rewrite@v0.1.0

# rewrite move imports of consul-topology to test/frameworks/topology
goimport-rewrite -p github.com/hashicorp/consul/consul-toplogy:github.com/hashicorp/consul/test/frameworks/toplogy .
```

---

> Maybe thats not desirable and we want to distinguish them in CI?

I'm not sure we really do need to distinguish them from e.g. the `consul-container` tests. I can't think of a substantial reason why they shouldn't live together, except that maybe they have a few helper functions that compete/collide.

One reason we might want to keep the tests separate from the framework is if this module is being imported by outside repos, which I believe it is (Cloud?). Having the tests inside the module means those others get a bunch of cruft they don't care about. Maybe not a big deal, but not super clean.

> go install github.com/mkeeler/goimport-rewrite@v0.1.0
Nice, I've been using `go-imports-rename` just because it was the first hit on Google :P